cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -138,4 +316,470 @@ private String getProgramScheduleId(Id.Program program, SchedulableProgramType p
+      }
+
+      if (lastNotification != null) {
+        // In all cases, when creating a schedule - either if it comes from the store or not,
+        // we want to pass it the last seen notification if it exists.
+        // This call will send the notification to all the active tasks. The ones which are active before
+        // that method is called will therefore see the notification twice. It is fine though.
+        received(lastNotification, null);
+      }
+      return true;
+    }
+
+    /**
+     * Suspend a scheduling task that is based on the data received by the stream referenced by {@code this} object.
+     */
+    public synchronized void suspendScheduleTask(Id.Program programId, SchedulableProgramType programType,
+                                                 String scheduleName) throws ScheduleNotFoundException {
+      String scheduleId = getScheduleId(programId, programType, scheduleName);
+      StreamSizeScheduleTask task = scheduleTasks.get(scheduleId);
+      if (task == null) {
+        throw new ScheduleNotFoundException(scheduleName);
+      }
+      if (task.suspend()) {
+        activeTasks.decrementAndGet();
+      }
+    }
+
+    /**
+     * Resume a scheduling task that is based on the data received by the stream referenced by {@code this} object.
+     */
+    public void resumeScheduleTask(Id.Program programId, SchedulableProgramType programType, String scheduleName)
+      throws ScheduleNotFoundException {
+      int activeTasksNow;
+      StreamSizeScheduleTask task;
+      synchronized (this) {
+        String scheduleId = getScheduleId(programId, programType, scheduleName);
+        task = scheduleTasks.get(scheduleId);
+        if (task == null) {
+          throw new ScheduleNotFoundException(scheduleName);
+        }
+        if (!task.resume()) {
+          return;
+        }
+        activeTasksNow = activeTasks.incrementAndGet();
+      }
+      if (activeTasksNow == 1) {
+        // There were no active tasks until then, that means polling the stream was disabled.
+        // We need to check if it is necessary to poll the stream at this time, if the last
+        // notification received was too long ago, or if there is no last seen notification
+        synchronized (lastNotificationLock) {
+          if (lastNotification == null ||
+            (lastNotification.getTimestamp() + pollingDelay <= System.currentTimeMillis())) {
+            // Resume stream polling
+            cancelPollingAndScheduleNext();
+            try {
+              StreamSize streamSize = getStreamEventsSize();
+              lastNotification = new StreamSizeNotification(streamSize.getTimestamp(), streamSize.getSize());
+            } catch (IOException e) {
+              LOG.debug("Ignoring stream events size polling after resuming schedule {} due to error", scheduleName, e);
+            }
+          }
+        }
+      }
+      task.received(lastNotification);
```

Comment:
Another remark. If you had two tasks that were both suspended, then you resume both of them. The first one will see activeTasksNow==1, poll for stream size and trigger the task. But the second one sees activeTasksNow==2, and it will not poll. Is there any other way to determine whether you have to poll when the schedule is resumed? Or perhaps you should always do it? Thoughts?